client/src/components/SearchResultsOffset.js

```diff
@@ -0,0 +1,167 @@
+import React from 'react'
+import { Box, Button, Paragraph, TextInput } from 'grommet'
+import { FormPrevious, FormNext } from 'grommet-icons'
+import { isOnlyNumbers } from '../helpers/isOnlyNumbers'
+import { useSearchResources } from '../hooks/useSearchResources'
+
+export const SearchResultsOffset = () => {
+  const {
+    query: { limit, offset },
+    response: { count },
+    setOffset,
+    goToSearchResults
+  } = useSearchResources()
+
+  const [enteredPageNumber, setEnteredPageNumber] = React.useState('')
+  const [
+    enteredPageNumberInRange,
+    setEnteredPageNumberInRange
+  ] = React.useState(false)
+
+  const safeOffset = parseInt(offset, 10) || 0
+  const safeCount = parseInt(count, 10) || 0
+  const safeLimit = parseInt(limit, 10)
+  const lastOffset =
+    safeCount && safeLimit > 0 ? Math.floor(safeCount / safeLimit) : 0
+
+  const startingPageSlices =
+    safeOffset <= 2 ? [0, 6] : [safeOffset - 2, safeOffset + 3]
+
+  const pageSlices = lastOffset - safeOffset <= 2 ? [-5] : startingPageSlices
```

Comment:
I think these calculations could be written a bit more clearly. I think an if statement would work better here than the ternary operator. `startingPageSlices` doesn't even need to be defined unless `lastOffset - safeOffset` is less than 2. It might be helpful to name `lastOffset - safeOffset` as a variable, because I'm not really sure what that number represents and why it's being compared to 2.

Reply:
i rewrote this for clarity